apiserver/pkg/http/client.go

```diff
@@ -638,29 +648,78 @@ func (krc *KuberayAPIServerClient) doDelete(deleteURL string) (*rpcStatus.Status
 }
 
 func (krc *KuberayAPIServerClient) executeRequest(httpRequest *http.Request, URL string) ([]byte, *rpcStatus.Status, error) {
-	response, err := krc.httpClient.Do(httpRequest)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to execute http request for url '%s': %w", URL, err)
-	}
-	defer func() {
-		if closeErr := response.Body.Close(); closeErr != nil {
-			klog.Errorf("Failed to close http response body because %+v", closeErr)
+	// Record the last error and status got
+	var lastErr error
+	var lastStatus *rpcStatus.Status
+
+	for attempt := 0; attempt < krc.maxRetry; attempt++ {
+		response, err := krc.httpClient.Do(httpRequest)
+		// do not retry for Go Error
```

Comment:
Could you please explain what do you mean by "Go Error"?
According to the [doc](https://pkg.go.dev/net/http#Client.Do): 
```
An error is returned if caused by client policy (such as CheckRedirect), or failure to speak HTTP (such as a network 
connectivity problem). A non-2xx status code doesn't cause an error.
```